### PR TITLE
Last Snapshot data to calculate power amount rank

### PR DESCRIPTION
**Pull request by aminlatifi**
Used last snapshot data to calculate expected project rank corresponding to power amount  (powerAmountRank)

Refs https://github.com/Giveth/giveth-dapps-v2/issues/1947

### CODE REVIEW

File: src/resolvers/projectPowerResolver.test.ts [hunk 1]
```diff
@@ -112,4 +117,180 @@ function projectPowersTestCases() {
...
+      userId: user1.id,
+      powerSnapshotId: snapshot.id,
+      balance: 10000,
+    });
+
+    await setPowerRound(roundNumber - 1);
+    await refreshProjectFuturePowerView();
+
+    let result = await axios.post(graphqlUrl, {
+      query: gePowerAmountRankQuery,
```
Comment (CarlosQ96) [inline src/resolvers/projectPowerResolver.test.ts]: Just noticed the Query is named wrongly it should be 'getPowerAmountRankQuery'
Ill push the renaming.

File: src/views/lastSnapshotProjectPowerView.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  @ViewColumn()
+  @Field(type => Float)
+  @Column('numeric', {
+    scale: 2,
+    transformer: new ColumnNumericTransformer(),
+  })
+  totalPower: number;
+
+  @ViewColumn()
+  // Not sure why queries return type was string!
```
Comment (CarlosQ96) [inline src/views/lastSnapshotProjectPowerView.ts:29]: I investigated but couldn't find a reason either.... super weird.

**Review by CarlosQ96 — APPROVED** [review]
Hey Amin thanks! Looks good to me. 

Only pushed some minor typo fixes. 

You can merge.

**Comment by aminlatifi** [comment]
Thank you @CarlosQ96

**Pull request merged**